lib/rebuild.js

```diff
@@ -39,16 +39,24 @@ const getFilterFn = args => {
     const spec = npa(arg)
     if (spec.type === 'tag' && spec.rawSpec === '')
       return spec
-    if (spec.type !== 'range' && spec.type !== 'version')
+
+    if (spec.type !== 'range' && spec.type !== 'version' && spec.type !== 'directory')
       throw new Error('`npm rebuild` only supports SemVer version/range specifiers')
+
     return spec
   })
+
   return node => specs.some(spec => {
-    const { version } = node.package
+    if (spec.type === 'directory')
+      return node.path === spec.fetchSpec
+
     if (spec.name !== node.name)
       return false
+
     if (spec.rawSpec === '' || spec.rawSpec === '*')
       return true
+
+    const { version } = node.package
```

Comment:
curious to see how a version-less `package.json` would behave 🤔 maybe add a test for it? (I would assume it might need to just default to `'*'`

Reply:
sorry, only now I realized this line has just been moved and is not necessarily part of supporting path as an argument (which is the original intent of the PR) in that case I'm fine with just adding a TODO comment for now 😊 

```suggestion
    // TODO: add tests for a package with missing version
    const { version } = node.package
```